network/drivers_wifi/nmcli.c

```diff
@@ -60,71 +60,79 @@ static void nmcli_stop(void *data) { }
 
 static bool nmcli_enable(void* data, bool enabled)
 {
-   /* semantics here are broken: nmcli_enable(..., false) is called
-    * on startup which is probably not what we want. */
+   int ret = 0;
 
-#if 0
    if (enabled)
-      pclose(popen("nmcli radio wifi on", "r"));
+      ret = system("nmcli radio wifi on");
    else
-      pclose(popen("nmcli radio wifi off", "r"));
-#endif
+      ret = system("nmcli radio wifi off");
 
-   return true;
+   return WIFEXITED(ret) && WEXITSTATUS(ret) == 0;
 }
 
 static bool nmcli_connection_info(void *data, wifi_network_info_t *netinfo)
 {
    FILE *cmd_file = NULL;
    char line[512];
+   bool connected = false;
 
-   if (!netinfo)
-      return false;
```

Comment:
When entering the `Wi-Fi` menu first, `nmcli_connection_info` is called with `NULL`. If we return `false` here, the menu doesn't show the `Disconnect` menu item even when the wifi is on and connected.